core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -282,6 +284,16 @@ private[coordinator] class GroupMetadata(val groupId: String, initialState: Grou
     pendingOffsetCommits ++= offsets
   }
 
+  def removeOffsets(topicPartition: TopicPartition) = {
+    if(pendingOffsetCommits.contains(topicPartition))
+      deletedPartitions += topicPartition
+    val removedOffsets = offsets.remove(topicPartition)
+    val removedOffsetMap = new mutable.HashMap[TopicPartition, OffsetAndMetadata]
```

Comment:
It's a little odd to repackage the value into a map. If we let `removeOffsets` only work with a single partition, it would be better to have `removeOffsets` return an `Option`. However, if you follow my suggestion above (to let `cleanupGroupMetadata` accept the full collection), then we could potentially also change this function to accept a collection of partitions, and the `Map` return type becomes more natural.